HELPS/ClientApp/src/types/components/UserTypes.ts

```diff
@@ -11,8 +11,15 @@ export interface UserDispatchProps {
     loadUserDetails: () => void
     updateUser: (user: Student) => void
     submit: () => void
+    loadAllUsers: () => void
+}
+
+export interface UserState {
+    loadAllUsers: Student
```

Comment:
You do not need loadAllUsers here, you will need a member called activeStudent or something similar which represents the selected student. 

The list of students will then have an onclick function that sets active student.

This active student member will then replace this.props.student in the getUserDetails function